javascript/organizing_your_javascript_code/webpack.md

```diff
@@ -1,24 +1,62 @@
+
+Similarly to how we will need loaders and rules for CSS and assets, we can use a plugin called `html-webpack-plugin` which will automatically build an HTML file in `dist` for us when we build our project. It will also then automatically add certain things to the HTML like our output bundle in a `<script>` tag, or the code to use a favicon if we configured it to use one.
+
+By default, it will just create this from a blank template, so the resulting HTML file will essentially be the usual boilerplate with our script and perhaps any other options we add in the configuration. If we had our own `dist/index.html` then it would be overwritten! In order to preserve our own HTML, we can tell it to use a template and provide a path to our own HTML file that is in `src`. You will see more examples of how to set up `html-webpack-plugin` appropriately in the assignment links below.
+
+An example of how to set the template option within your webpack configuration file is as follows:
+
+~~~javascript
+// webpack.config.js
+
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+// ...
+    plugins: [
+        new HtmlWebpackPlugin({
+            template: './src/template.html',
+        }),
+    ],
+// ...
+~~~
 
 ### Assignment
 
 <div class="lesson-content__panel" markdown="1">
 
 1. Code along with [this tutorial](https://webpack.js.org/guides/asset-management/) to see examples of using webpack to manage your website's assets.
-2. Code along with [this tutorial](https://webpack.js.org/guides/output-management/) to learn how to let webpack automatically manage your index.html and insert your bundle into the page for you!
-3. Finally, the first part of [this tutorial](https://webpack.js.org/guides/development/) talks about source maps, a handy way to track down which source file (index.js, a.js, b.js) an error is coming from when you use webpack to bundle them together. This is essential to debugging bundled code in your browser's DevTools. If the error comes from b.js the error will reference that file instead of the bundle. It also walks through an example of the `--watch` feature you _definitely_ should have taken note of above.
-4. You don't need to do the rest of the webpack tutorials at this time, but take a look at what's offered on the sidebar of their [guides page](https://webpack.js.org/guides/). There are several sweet features that you might want to use in future projects such as code-splitting, lazy-loading, and tree-shaking. Now that you have a handle on webpack's configuration system adding these features is as easy as using the right plugins and loaders!
+2. Watch [this video](https://rapidevelop.org/webpack/setup-html-webpack-plugin) which demonstrates how to install `html-webpack-plugin` and set different options on it (note that some of the options shown are redundant as they are already the [default settings](https://github.com/jantimon/html-webpack-plugin#options)).
```

Comment:
Just checked out the link. Didn't have time to watch the whole video but it looks like the video steps are reproduced in writing. Can you confirm that's the case? If so, this feels like it's a good add. Just want to make sure this resource is fully accessible. :)

Reply:
The writing doesn't show any examples of code, only describes things but given that the template option is the most important, an example of it is already shown in the proposed lesson content, and the plugin docs have plenty of examples, I think that yeah, I could probably adjust what's written here very slightly so it's not strictly "watch the video".